src/components/Api.js

```diff
@@ -0,0 +1,220 @@
+export default class Api {
+  constructor(options) {
+    this._url = options.url;
+    this.authorization = options.authorization;
+  }
+
+  getCurrentUser(endPoint, DOMLocations) {
+    return fetch(`${this._url}${endPoint}`, {
+      method: "GET",
+      headers: {
+        authorization: this.authorization,
+      },
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Something went wrong ${res.status}`);
+      })
+      .then((result) => {
+        console.log(result);
+        console.log("umm now what??");
+        DOMLocations.name.textContent = result.name;
+        DOMLocations.about.textContent = result.about;
+        DOMLocations.avatar.src = result.avatar;
+        // const profileData = {
+        //   name: result.name,
+        //   job: result.about,
+        // };
+        // return profileData;
+      })
+      .catch((err) => console.error(err));
```

Comment:
Needs correction: Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```

Reply:
done